src/Analyzer.Utilities/Extensions/IEnumerableOfIMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,92 @@
+                 int j = 0;
+
+                if (!trailingOnly && candidateMethod.Parameters.First().Type.Equals(expectedParameterType))
+                {
+                     // If expectedParameterType is the first parameter then the parameters to compare in candidateMethod against selectedOverload
+                     // is offset by 1
+                     j = 1;
+                }
+                else if (!candidateMethod.Parameters.Last().Type.Equals(expectedParameterType))
+                {
+                     // expectedParameterType is neither the first parameter nor the last parameter
+                     return false;
+                }
+
+                for (int i = 0; i < selectedOverload.Parameters.Count(); i++, j++)
+                {
+                    if (!selectedOverload.Parameters[i].Type.Equals(candidateMethod.Parameters[j].Type) ||
+                        selectedOverload.Parameters[i].IsParams != candidateMethod.Parameters[j].IsParams)
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            });
+        }
+
+        /// <summary>
+        /// Returns a list of method symbols from a given list of the method symbols, which has its parameter type as
+        /// expectedParameterType as its last parameter in addition to matching all the other parameter types of the 
+        /// selectedOverload method symbol
+        /// </summary>
+        /// <param name="methods">List of <see cref="IMethodSymbol"/> to scan for possible overloads</param>
+        /// <param name="selectedOverload"><see cref="IMethodSymbol"/> that is currently picked by the user</param>
+        /// <param name="expectedParameterType"><see cref="INamedTypeSymbol"/> type of the leading parameter or the trailing parameter</param>
+        public static IEnumerable<IMethodSymbol> GetMethodOverloadsWithDesiredParameterAtTrailing(
+             this IEnumerable<IMethodSymbol> methods,
+             IMethodSymbol selectedOverload,
+             INamedTypeSymbol expectedParameterType)
+        {
+            return GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(methods, selectedOverload, expectedParameterType, true);
+        }
+
+        /// <summary>
+        /// Return all the method symbols with the name matching the displayName case-insensitively
+        /// </summary>
+        /// <param name="stringFormatMembers"></param>
+        /// <param name="displayName"></param>
+        /// <returns></returns>
+        public static IMethodSymbol GetSingleOrDefaultMemberWithName(this IEnumerable<IMethodSymbol> stringFormatMembers, string displayName)
+        {
+             return stringFormatMembers?.Where(member => string.Equals(member.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat), displayName, StringComparison.OrdinalIgnoreCase)).SingleOrDefault();
```

Comment:
@mavasani  This was discussed in the earlier PR as well. Though the format used here is named CSharpErrorMessageFormat, this has got nothing to do with CSharp. This is just one other SymbolDisplayFormat and it is used by CSharp. The reason to use this instead of the each symbol's default display format is, the default display for VB symbols are `Public Property Name as String` which doesn't fit well in the error messages.

Both the base type analyzer and the helper is in language agnostic part of the code so I feel that there is no harm in keeping it here.